packages/design-system/src/components/popup/utils/getOffset.js

```diff
@@ -128,7 +133,12 @@ export function getOffset({
   const { x: spacingH = 0, y: spacingV = 0 } = spacing || {};
 
   // Horizontal
-  const offsetX = getXOffset(placement, spacingH, anchorRect, dockRect, isRTL);
+  const offsetX = () => {
+    if (resetXOffset && popupRect?.left <= 0) {
+      return isRTL ? width : 0;
+    }
+    return getXOffset(placement, spacingH, anchorRect, dockRect, isRTL);
+  };
```

Comment:
Since tooltips update the placement of a popup, and since some popups can have negative left values, I created this `resetXOffset` flag to limit when we would potentially reset the offset to the width
![Screen Shot 2022-03-14 at 1 48 47 PM](https://user-images.githubusercontent.com/1820266/158249701-50d47b77-92a1-47fb-8d29-ea6a450f0c0c.png)
 of the popup. This forces the popup to start at the left edge of the screen in in RTL.